README


This archive contains the distribution Dist-Zilla-Plugin-PromptIfStale,
version 0.033:

  Check at build/release time if modules are out of date

This software is copyright (c) 2013 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.029.



Changes
Revision history for Dist-Zilla-Plugin-PromptIfStale

0.033     2015-01-03 20:27:49Z
          - bump Module::Metdata prereq for $VERSION parsing (RT#101095)

0.032     2014-11-22 03:36:00Z
          - better way of avoiding warnings in perl 5.21.6, while still
            allowing this dist to install
          - switch from cpanidx to cpanmetadb index

0.031     2014-11-17 02:51:40Z
          - add a dynamic test prereq to avoid new warnings added to perl 5.21.6

0.030     2014-11-13 06:48:50Z
          - make prompt tests unfailable on perls older than 5.16, to work around an
            apparent issue with IO::Pty

0.029     2014-11-08 23:55:57Z
          - ensure @INC does not become mangled during tests
          - use newer Dist::Zilla::Util::AuthorDeps to avoid using Path::Class

0.028     2014-10-12 18:51:44Z
          - added extra diagnostics on filehandle states when tests fail

0.027     2014-09-24 01:58:26Z
          - fix tty handling in tests to resolve filehandle issues on linux
            (RT#99074)

0.026     2014-09-12 03:24:50Z
          - fix tests (new in 0.025) that fail when stdin is not a tty (e.g.
            piping module names to cpanm)

0.025     2014-09-09 02:43:51Z
          - in "fatal" mode, also print the detailed list of what is stale and
            why, before dying with an advisory to run 'cpanm'
          - only issue a warning, rather than prompting, when being run
            non-interactively

0.024     2014-07-23 20:13:17Z
          - switch from using JSON to JSON::MaybeXS
          - avoid reporting core (non-duallifed) modules as being stale, since
            they cannot be updated without updating perl itself

0.023     2014-05-15 20:38:19Z
          - 'dzil stale' now respects all plugins' skip options
          - improved debug logging
          - cache more data for efficiency when multiple plugins, options are
            used at once
          - new "check_authordeps" option

0.022     2014-04-19 08:28:04Z
          - "dzil stale" now falls back even harder to parsing authordeps out
            of dist.ini when performing the build blew up (such as due to an
            unsatisfied develop prereq)
          - new sister plugin, [EnsureNotStale], which aborts instead of
            prompting (this is for you, Getty - sorry it's too late)

0.021     2014-03-23 05:40:38Z
          - 'dzil stale' properly falls back to fetching authordeps when a
            plugin is not at the required version

0.020     2014-03-01 06:49:02Z
          - fix "dzil stale --all" when no [PromptIfStale] plugins are
            configured for the distribution
          - dzil stale now respects the --verbose flag
          - dzil stale now does the equivalent of 'dzil authordeps' if some
            plugins are missing, rather than dying -- you can now use 'dzil
            stale' as your one stop shop for ensuring all requirements are
            fulfilled

0.019     2014-02-28 04:34:49Z
          - fix occasional test failures on win32 due to varying directory
            separators in filenames

0.018     2014-02-23 23:29:22Z
          - new --all option for "dzil stale"

0.017     2014-02-12 05:20:07Z
          - fix uninitialized warning when 02packages.details.txt cannot be
            downloaded

0.016     2014-02-09 03:41:16Z
          - new "dzil stale" command

0.015     2013-12-10 23:38:52Z, a.k.a. "The Getty Release"
          - new 'fatal' option, to avoid prompting
          - new 'index_base_url' option, for using different CPAN indexes

0.014     2013-11-29 05:53:17Z
          - new 'skip' option

0.013     2013-11-15 04:19:33Z
          - more diagnostics printed in failing tests

0.012     2013-11-10 19:27:11Z (TRIAL RELEASE)
          - no longer loading each module being checked in order to get its
            $VERSION
          - fix logic for choosing when to pull down a copy of
            02packages.details.txt (now only doing so when otherwise we would
            make six or more HTTP hits to PAUSE)
          - more diagnostics printed in failing tests

0.011     2013-11-10 01:54:55Z
          - honour check_all_prereqs setting at release time (previously
            behaviour was always as if it was on); thanks David Golden!

0.010     2013-11-03 20:22:01Z (TRIAL RELEASE)
          - more diagnostics printed in failing tests

0.009     2013-11-02 21:15:16Z
          - work around issues with EUMM-bundled version.pm by properly
            fudging a VERSION during testing (affected perls before 5.13.9)

0.008     2013-10-11 01:47:09Z
          - now also checks the configure and build prerequisites

0.007     2013-10-02 03:58:16Z
          - when killing a build after being prompted, include a line that can
            be copy-pasted to install all the missing modules via cpanm
          - speed up index querying when checking for 6 or more modules at
            once by pulling down a copy of 02packages.details.txt.gz

0.006     2013-09-24 04:36:43Z
          - diagnostic message added before checks begin, so the user isn't
            wondering why the build/release appears to hang (querying the
            index can take a little bit of time if there are a lot of modules
            to check)

0.005     2013-09-16 00:05:07Z
          - now only prompting once (per phase) for all discrepancies found
          - no longer prompting for the same module twice, if there are
            multiple instances of the plugin running

0.004     2013-09-11 01:39:02Z
          - New "check_all_prereqs" option
          - warnings tests bypassed during installation

0.003     2013-09-05 07:45:52Z
          - gracefully handle checking for a module that is not installed
          - support setting both the check_all_plugins and module configs at
            once

0.002     2013-08-13 01:04:29Z
          - fix documentation that referred to an option by the wrong name
          - inverted prompt, from "abort build?" to "continue anyway?" (with

Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;

use 5.010;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Check at build/release time if modules are out of date",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-PromptIfStale",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010",
  "NAME" => "Dist::Zilla::Plugin::PromptIfStale",
  "PREREQ_PM" => {
    "CPAN::DistnameInfo" => 0,
    "Cwd" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Util::AuthorDeps" => "5.021",
    "Encode" => 0,
    "HTTP::Headers" => 0,
    "HTTP::Tiny" => 0,
    "List::MoreUtils" => 0,
    "List::Util" => "1.33",
    "Module::CoreList" => "3.10",
    "Module::Metadata" => "1.000023",
    "Module::Path" => "0.18",
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "MooseX::Types::Moose" => 0,
    "Parse::CPAN::Packages::Fast" => 0,
    "Path::Tiny" => 0,
    "Try::Tiny" => 0,
    "YAML::Tiny" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::App::Tester" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "Module::Runtime" => 0,
    "Moose::Util" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "if" => 0,
    "lib" => 0
  },
  "VERSION" => "0.033",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::DistnameInfo" => 0,
  "Cwd" => 0,
  "Dist::Zilla::App" => 0,
  "Dist::Zilla::App::Tester" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::BeforeRelease" => 0,
  "Dist::Zilla::Util::AuthorDeps" => "5.021",
  "Encode" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "HTTP::Headers" => 0,
  "HTTP::Tiny" => 0,
  "List::MoreUtils" => 0,
  "List::Util" => "1.33",
  "Module::CoreList" => "3.10",
  "Module::Metadata" => "1.000023",
  "Module::Path" => "0.18",
  "Module::Runtime" => 0,
  "Moose" => 0,
  "Moose::Util" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "MooseX::Types::Moose" => 0,
  "Parse::CPAN::Packages::Fast" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "Try::Tiny" => 0,
  "YAML::Tiny" => 0,
  "if" => 0,
  "lib" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);

# inserted by Dist::Zilla::Plugin::DynamicPrereqs 0.009
$WriteMakefileArgs{TEST_REQUIRES}{'IO::Pty'} = $FallbackPrereqs{'IO::Pty'} = '0'
  if $^O ne 'MSWin32';



unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);





Build.PL
# This Build.PL for Dist-Zilla-Plugin-PromptIfStale was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.006
use strict;
use warnings;

if (eval 'use Module::Build::Tiny 0.038; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-PromptIfStale was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.007.
    use strict;
    use warnings;

    use 5.010;
    # use Module::Build::Tiny 0.038;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.038"
      },
      "dist_abstract" => "Check at build/release time if modules are out of date",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-PromptIfStale",
      "dist_version" => "0.025",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::PromptIfStale",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Cwd" => 0,
        "Dist::Zilla::App" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::AfterBuild" => 0,
        "Dist::Zilla::Role::BeforeBuild" => 0,
        "Dist::Zilla::Role::BeforeRelease" => 0,
        "Dist::Zilla::Util::AuthorDeps" => 0,
        "HTTP::Tiny" => 0,
        "JSON::MaybeXS" => 0,
        "List::MoreUtils" => 0,
        "List::Util" => "1.33",
        "Module::CoreList" => "3.10",
        "Module::Metadata" => 0,
        "Module::Path" => 0,
        "Moose" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "MooseX::Types::Moose" => 0,
        "Parse::CPAN::Packages::Fast" => 0,
        "Path::Class" => 0,
        "Path::Tiny" => 0,
        "Try::Tiny" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.010",
        "strict" => 0,
        "version" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "Dist::Zilla::App::Tester" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::pushd" => 0,
        "Module::Runtime" => 0,
        "Moose::Util" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "if" => 0,
        "lib" => 0
      }
    );


    my %fallback_build_requires = (
      "Dist::Zilla::App::Tester" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::pushd" => 0,
      "Module::Runtime" => 0,
      "Moose::Util" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "if" => 0,
      "lib" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
